app/models/commit_monitor_branch.rb

```diff
@@ -31,4 +31,19 @@ def last_commit_uri
   def pr_number
     MiqToolsServices::MiniGit.pr_number(name) if pull_request?
   end
+
+  def write_github_comment(message, header, continuation_header = nil, replace = true)
+    return unless pull_request?
+
+    message_builder = MiqToolsServices::Github::MessageBuilder.new(header + message, continuation_header)
```

Comment:
This method was rewritten as the caller should pass in the header as the tag + first line.  The new code will look similar to your recommendation.